pandas/tests/indexes/test_numeric.py

```diff
@@ -944,6 +944,11 @@ def test_constructor(self):
         res = Index(np.array([-1, 2 ** 63], dtype=object))
         tm.assert_index_equal(res, idx)
 
+        # https://github.com/pandas-dev/pandas/issues/29526
+        idx = UInt64Index([1, 2 ** 63 + 1], dtype=np.uint64)
```

Comment:
I don't think that this should require passing in the `dtype` in order to not lose precision, and it currently looks like that's the case:
```python
In [1]: import pandas as pd; pd.__version__
Out[1]: '0.26.0.dev0+947.gb9b462c01'

In [2]: pd.UInt64Index([1, 2**63 + 1], dtype='uint64')
Out[2]: UInt64Index([1, 9223372036854775809], dtype='uint64')

In [3]: pd.UInt64Index([1, 2**63 + 1])
Out[3]: UInt64Index([1, 9223372036854775808], dtype='uint64')
```

Reply:
nevermind, missed your previous comment on this matter